workflow-bot-app/src/test/java/com/symphony/bdk/workflow/engine/camunda/monitoring/repository/ActivityCmdaApiQueryRepositoryTest.java

```diff
@@ -0,0 +1,111 @@
+package com.symphony.bdk.workflow.engine.camunda.monitoring.repository;
+
+import static org.assertj.core.api.BDDAssertions.then;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.ArgumentMatchers.contains;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.symphony.bdk.workflow.converter.ObjectConverter;
+import com.symphony.bdk.workflow.monitoring.repository.domain.ActivityInstanceDomain;
+
+import org.camunda.bpm.engine.HistoryService;
+import org.camunda.bpm.engine.history.HistoricActivityInstance;
+import org.camunda.bpm.engine.history.HistoricVariableInstance;
+import org.camunda.bpm.engine.history.NativeHistoricVariableInstanceQuery;
+import org.camunda.community.mockito.QueryMocks;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@ExtendWith(MockitoExtension.class)
+class ActivityCmdaApiQueryRepositoryTest {
```

Comment:
Given that the objectConverter is mocked, I'm wondering if we really need to mock the repositories here? 

Let's say we have `objectConverter.convertCollection(historyService.doSomething())`, if we mock convertCollection result, we don't really need to mock historiService.doSomething()..

Or am I wrong?

Reply:
well, it is partially true. the repository mock is needed, because we need the mock to return a list anyway, it s just we don't necessary need the mock from camunda, it is used here because it s quite handy, and we know that which camunda api is used in the repository class.